src/main/daml/Daml/Finance/Instrument/Swap/Asset/Instrument.daml

```diff
@@ -84,6 +82,9 @@ template Instrument
       view = Claim.View with acquisitionTime = dateToDateClockTime $ daysSinceEpochToDate 0
       getClaims Claim.GetClaims{actor} = do
         -- get the initial claims tree (as of the swap's acquisition time)
+        assertMsg "the total weight must be 1" $
+          foldl (\acc u -> u.weight + acc) 0.0 underlyings == 1.0
```

Comment:
Can use [sum](https://docs.daml.com/daml/stdlib/DA-Foldable.html#function-da-foldable-sum-87024)?

Reply:
How would that work? Seems it only takes one argument. Would I have to make the Underlying additive and somehow specify what it is that should be summed? There is also an initial price in there, which should not be used.

Reply:
right, we are applying the function to the `Underlying`. In that case, `foldMap (.weight)` should do the trick